pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -68,6 +68,12 @@ func (p *ldapProvider) loginUser(credential *v32.BasicLogin, config *v3.LdapConf
 		return v3.Principal{}, nil, httperror.WrapAPIError(err, httperror.ServerError, "server error while authenticating")
 	}
 
+	/* Rebind as read-only / bind user account for further queries */
+	err = ldap.AuthenticateServiceAccountUser(serviceAccountPassword, serviceAccountUserName, "", lConn)
+	if err != nil {
+		return v3.Principal{}, nil, err
+	}
+
```

Comment:
Does the Bind using userDN fail with any error code? If it does not error out why do we need to rebind using the serviceAccount?

Reply:
I got an "OpenLDAP: Result Code 32 No Such Object" when I tried to configure authentication. More info can be found in these related issues: #35259, #17854 and #25165.

Reply:
Do you get the same error code if you use the user's cred and bind using ldapsearch tool or any other tool that does nt involve Rancher?

Reply:
Can we add this re-bind as read-only / bind user account in case of an error?